### PR TITLE
UIU-1492 Custom fields: Create UI permissions for Settings > Users > Custom Fields CRUD

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Purpose

- Apply permission restrictions to Users > Custom Fields

### CODE REVIEW

File: src/routes/EditCustomFields.js [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
       redirectToView={redirectToView}
+      permissions={permissions}
     />
   );
 };
 
 EditCustomFields.propTypes = propTypes;
 
-export default EditCustomFields;
+export default withStripes(EditCustomFields);
```
Comment (MaksymDryha) [inline src/routes/EditCustomFields.js]: please, use `useStripes` hook

File: src/index.js [hunk 2]
```diff
@@ -270,7 +271,15 @@ class UsersRouting extends Component {
         <Route path={`${base}/:id/patronblocks/edit/:patronblockid`} component={Routes.PatronBlockContainer} />
         <Route path={`${base}/:id/patronblocks/create`} component={Routes.PatronBlockContainer} />
         <Route path={`${base}/create`} component={Routes.UserEditContainer} />
-        <Route path={`${base}/custom-fields/edit`} exact component={Routes.EditCustomFields} />
+        <Route
+          path={`${base}/custom-fields/edit`}
+          exact
+          render={(props) => (
+            <IfPermission perm="ui-users.settings.customfields.edit">
```
Comment (MaksymDryha) [inline src/index.js]: We shouldn't wrap the route component in `IfPermission`. Instead, please implement the logic of redirection to the root page of users app if the user doesn't have `customfields.edit` permission. This should be implemented by `Routes.EditCustomFields`.

File: src/settings/CustomFieldsSettings.js [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
       redirectToEdit={redirectToEdit}
+      permissions={permissions}
     />
   );
 };
 
 CustomFieldsSettings.propTypes = propTypes;
 
-export default CustomFieldsSettings;
+export default withStripes(CustomFieldsSettings);
```
Comment (MaksymDryha) [inline src/settings/CustomFieldsSettings.js]: the same about `useStripes` hook

File: src/routes/EditCustomFields.js [hunk 4]
```diff
@@ -12,15 +13,28 @@ const propTypes = {
...
   };
 
+  const permissions = {
+    canView: stripes.hasPerm('ui-users.settings.customfields.view'),
+    canEdit: stripes.hasPerm('ui-users.settings.customfields.edit'),
+    canDelete: stripes.hasPerm('ui-users.stripes.customfields.delete'),
+  };
+
+  if (!permissions.canView) {
+    history.push('/users');
```
Comment (MaksymDryha) [inline src/routes/EditCustomFields.js]: I guess `history.replace` should be used

**Comment by zburke** [comment]
Sorry I missed this when I pushed #1193. I filed [UIU-1521](https://issues.folio.org/browse/UIU-1521) because at the time, all open PRs, including this one, were listed as unmergeable.

**Pull request merged**